File: datasets/JPlag-Java/to-kotlin-intellij/Compact.kt
package java

object Compact {
    fun tryCount(a: ArrayList<Int>, n: Int): Int {
        if (a.isEmpty()) return Int.MAX_VALUE

        var count = 0

        var i = 0
        while (i + 1 < a.size) {
            if (a[i] + 1 != a[i + 1]) count++
            i++
        }

        if (a[0] != 0) count++

        if (a[a.size - 1] != n - 1) count++

        return count
    }
}

File: datasets/JPlag-Java/to-typescript-convertai/Compact.ts
import { ArrayList } from 'java.util';

export class Compact {
    static tryCount(a: ArrayList<number>, n: number): number {
        if (a.isEmpty()) {
            return Number.MAX_SAFE_INTEGER;
        }

        let count = 0;

        for (let i = 0; i + 1 < a.size(); i++) {
            if (a.get(i) + 1 !== a.get(i + 1)) {
                count++;
            }
        }

        if (a.get(0) !== 0) {
            count++;
        }

        if (a.get(a.size() - 1) !== n - 1) {
            count++;
        }

        return count;
    }
}

